spire/db.py

```diff
@@ -36,4 +38,15 @@ def yield_connection_from_env() -> Session:
         session.close()
 
 
+@contextmanager
+def yield_redis_env_ctx():
+    try:
+        redis_client = redis.Redis().from_url(
```

Comment:
I don't like that you are creating a new connection on every `POST /humbug/reports` request.

You should create a `ConnectionPool` object global: https://redis-py.readthedocs.io/en/stable/#redis.ConnectionPool

Add a method called `close_redis_connections` which closes the connections in the pool and an `app.on_event("shutdown")(db.close_redis_connections)` in `api.py`.

The `yield_redis_connection_from_env` generator should probably call `get_connection` on the connection pool. Or just create a `Redis` instance, whatever you want. You probably shouldn't close the connection in the `finally` statement of the yield since the connections are managed by the pool. Maybe there is a connection release mechanism to release the connection back to the pool?

Some stackoverflow post: https://stackoverflow.com/questions/31663288/how-do-i-properly-use-connection-pools-in-redis